javatests/arcs/schematests/Arcs.kt

```diff
@@ -0,0 +1,185 @@
+@file:Suppress("EXPERIMENTAL_IS_NOT_ENABLED")
+
+package arcs.schematests
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import arcs.android.sdk.host.AndroidHost
+import arcs.android.storage.database.AndroidSqliteDatabaseManager
+import arcs.core.allocator.Allocator
+import arcs.core.common.toArcId
+import arcs.core.host.EntityHandleManager
+import arcs.core.host.SchedulerProvider
+import arcs.core.host.toRegistration
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import arcs.jvm.host.ExplicitHostRegistry
+import arcs.jvm.host.JvmSchedulerProvider
+import arcs.jvm.util.JvmTime
+import arcs.sdk.android.storage.ServiceStoreFactory
+import arcs.sdk.android.storage.service.ConnectionFactory
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.coroutines.EmptyCoroutineContext
+
+
+/** Container to own the allocator and start the long-running arc. */
+class Arcs(
+    private val context: Context,
+    // A test [ConnectionFactory] can be provided here under test.
+    // In production, leave this parameter as null. Arcs will provide a default implementation.
+    connectionFactory: ConnectionFactory? = null
+) {
+    val schedulerProvider = JvmSchedulerProvider(EmptyCoroutineContext)
+
+
+    private val fakeLifecycleOwner = object : LifecycleOwner {
+        private val lifecycle = LifecycleRegistry(this)
+        override fun getLifecycle() = lifecycle
+    }
+
+
+    val arcHost = ArcHost(
+        context,
+        fakeLifecycleOwner.lifecycle,
+        schedulerProvider,
+        connectionFactory
+    )
+
+    private lateinit var allocator: Allocator
+
+    /**
+     * Start the arc containing the particles that provide the persistence. This should be run
+     * one time per instance, via [startIfNot].
+     */
+    private suspend fun startArc() {
+        /** This should probably live in an initializer somewhere. */
+        val hostRegistry = ExplicitHostRegistry().apply {
```

Comment:
The only downside of this is that it doesn't test the way ArcHosts are embedded in services, and interactions with Android lifecycle events.  There is flakyness associated with Service embedded ArcHosts that doesn't appear in non-Service embedded ArcHosts.

Reply:
We're not currently using Service-embedded ArcHosts. The project I'm trying to reflect in this PR is using a pattern that looks more or less like this one, with the `Arcs` object being provided by Dagger as a singleton, and then being passed to a component to use as a persistence layer.